### PR TITLE
Added logic to allow wildcards at the end of paths

### PR DESCRIPTION
**Pull request by hc-jordenlowe**
Added logic to allow wildcards at the end of paths

### CODE REVIEW

File: Fabric.Platform.Auth/AuthorizationMiddleware.cs [hunk 1]
```diff
@@ -33,5 +34,28 @@ public static AppFunc Inject(AppFunc next, string[] requiredScopes, string[] all
                 return Task.CompletedTask;
             };
         }
+
+        private static bool HasPath(string[] allowedPaths, string pathToCheck)
+        {
+            foreach(var path in allowedPaths)
+            {
```
Comment (bsmithvt) [inline Fabric.Platform.Auth/AuthorizationMiddleware.cs]: Maybe move the asterisk to a constant?

File: Fabric.Platform.Auth/AuthorizationMiddleware.cs [hunk 2]
```diff
@@ -33,5 +34,28 @@ public static AppFunc Inject(AppFunc next, string[] requiredScopes, string[] all
...
+                if(path.Last() == '*')
+                {
+                    if(pathToCheck.StartsWith(path.TrimEnd('*')))
+                    {
+                        return true;
+                    }
+                }
+                else
+                {
```
Comment (bsmithvt) [inline Fabric.Platform.Auth/AuthorizationMiddleware.cs]: Do we want case this to be case-insensitive?

**Review by bsmithvt — APPROVED** [review]

File: Fabric.Platform.Auth/AuthorizationMiddleware.cs [hunk 3]
```diff
@@ -33,5 +34,28 @@ public static AppFunc Inject(AppFunc next, string[] requiredScopes, string[] all
...
             };
         }
+
+        private static bool HasPath(string[] allowedPaths, string pathToCheck)
+        {
+            foreach(var path in allowedPaths)
+            {
+                if(path.Last() == '*')
+                {
+                    if(pathToCheck.StartsWith(path.TrimEnd('*')))
```
Comment (michaelvidal24) [inline Fabric.Platform.Auth/AuthorizationMiddleware.cs]: Do we want to combine these if statements?

**Review by michaelvidal24 — APPROVED** [review]

File: Fabric.Platform.Auth/AuthorizationMiddleware.cs [hunk 4]
```diff
@@ -33,5 +34,28 @@ public static AppFunc Inject(AppFunc next, string[] requiredScopes, string[] all
                 return Task.CompletedTask;
             };
         }
+
+        private static bool HasPath(string[] allowedPaths, string pathToCheck)
+        {
```
Comment (bsmithvt) [inline Fabric.Platform.Auth/AuthorizationMiddleware.cs]: I think we can consolidate this into just 1 statement?

`return allowedPaths.Contains(pathToCheck)  || allowedPaths.Any(path => pathToCheck.StartsWith(path.TrimEnd('*'));`

Am I missing something? Would have to write some unit tests to confirm.

**Review by jason-qq — APPROVED** [review]

File: Fabric.Platform.Auth/AuthorizationMiddleware.cs (hunk 1 above)
Comment (hc-jordenlowe) [reply Fabric.Platform.Auth/AuthorizationMiddleware.cs]: PathSymbol.AnyNumberOfCharacters, I did that.

File: Fabric.Platform.Auth/AuthorizationMiddleware.cs (hunk 3 above)
Comment (hc-jordenlowe) [reply Fabric.Platform.Auth/AuthorizationMiddleware.cs]: done

File: Fabric.Platform.Auth/AuthorizationMiddleware.cs (hunk 2 above)
Comment (hc-jordenlowe) [reply Fabric.Platform.Auth/AuthorizationMiddleware.cs]: Before, the logic used Contains, which is case-sensitive.  I dont think there is a need for making it less restrictive.  Im going to skip it for now, but if there is a good reason we can always change it later.

**Pull request merged**